src/libs/ajax.js

```diff
@@ -10,6 +10,165 @@ import { ajaxOverridesStore, knownBucketRequesterPaysStatuses, requesterPaysProj
+    destroyedDate: null,
+    dateAccessed: '2020-07-16T18:42:09.784Z'
+  },
+  size: 500,
+  diskType: 'pd-standard',
+  blockSize: 4096
+}
+
+const deletingDisk = {
+  id: 22,
+  googleProject: 'general-dev-billing-account',
+  zone: 'us-central1-a',
+  name: 'fakeDisk2',
+  status: 'Deleting',
+  auditInfo: {
+    creator: 'b.adm.firec@gmail.com',
+    createdDate: '2020-07-17T18:42:14.941807Z',
+    destroyedDate: null,
+    dateAccessed: '2020-07-16T18:42:09.784Z'
+  },
+  size: 505,
+  diskType: 'pd-standard',
+  blockSize: 4096
+}
+
+const runtimeDetails = {
+  runtimeImages: [
+    { imageType: 'Jupyter', imageUrl: 'us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:1.0.3', timestamp: '2020-07-17T15:07:52.241Z' },
+    { imageType: 'Proxy', imageUrl: 'broadinstitute/openidc-proxy:2.3.1_2', timestamp: '2020-07-17T15:07:52.241Z' },
+    {
+      imageType: 'VM', imageUrl: 'projects/broad-dsp-gcr-public/global/images/custom-leo-image-dataproc-1-4-15-debian9-2020-07-02',
+      timestamp: '2020-07-17T15:07:53.046Z'
+    },
+    { imageType: 'Welder', imageUrl: 'us.gcr.io/broad-dsp-gcr-public/welder-server:5426cf0', timestamp: '2020-07-17T15:07:52.241Z' }
+  ],
+  jupyterUserScriptUri: null
+}
+
+// eslint-disable-next-line no-unused-vars
+const pdOverrides = _.mapValues(({ runtimes, disks }) => {
+  const stub = v => () => async (url, { method = 'GET', body }) => {
+    console.log({ method, url, body: body && JSON.parse(body) })
+    await Utils.delay(1000)
+    return new Response(JSON.stringify(v))
+  }
+  return [
+    { filter: { url: /v1\/runtimes\?/ }, fn: stub(runtimes) },
+    { filter: { url: /v1\/disks\?/ }, fn: stub(disks) },
+    { filter: { url: /v1\/disks\/.+\/.+$/, method: 'PATCH' }, fn: stub({}) },
+    { filter: { url: /v1\/disks\/.+\/.+$/, method: 'DELETE' }, fn: stub({}) },
+    // the below GET call doesn't include disk details, but we don't need them at the moment
+    { filter: { url: /v1\/disks\/.+\/.+$/, method: 'GET' }, fn: stub(disks[0]) },
+    { filter: { url: /v1\/runtimes\/.+\/.+$/, method: 'GET' }, fn: stub({ ...runtimes[0], ...runtimeDetails }) },
+    { filter: { url: /v1\/runtimes\/.+\/.+$/, method: 'POST' }, fn: stub({}) },
+    { filter: { url: /v1\/runtimes\/.+\/.+$/, method: 'PATCH' }, fn: stub({}) },
+    { filter: { url: /v1\/runtimes\/.+\/.+$/, method: 'DELETE' }, fn: stub({}) }
+  ]
+}, {
+  nothing: { runtimes: [], disks: [] },
+  gce: { runtimes: [gceRuntime], disks: [] },
+  dataproc: { runtimes: [dataprocRuntime], disks: [] },
+  dataprocCluster: { runtimes: [dataprocClusterRuntime], disks: [] },
+  disk: { runtimes: [], disks: [disk] },
+  diskAndDeletingDisk: { runtimes: [], disks: [disk, deletingDisk] },
+  gceAndDisk: { runtimes: [gceRuntime], disks: [disk] },
+  gceAndAttachedDisk: { runtimes: [gceRuntimeWithPd], disks: [disk] },
+  dataprocAndDisk: { runtimes: [dataprocRuntime], disks: [disk] },
+  attachedDiskEdgeCase: { runtimes: [gceRuntimeWithPd, dataprocRuntime], disks: [disk] }
+})
+
```

Comment:
Yes. I recognize that these are likely to get out-of-date over time and could end up being a detriment instead of a benefit. However, I am sad that we'll have to recreate these any time we make changes to the panel in order to test it. As with the debug panels, there is a bit of learning built in to the existence of these cases, and we rely on that kind of learning to know what to manually test when making changes.